pkg/cmd/openshift-apiserver/openshiftapiserver/openshift_apiserver.go

```diff
@@ -509,6 +513,48 @@ func (c *completedConfig) withUserAPIServer(delegateAPIServer genericapiserver.D
 	return server.GenericAPIServer, &legacyStorageVersionMutator{version: userapiv1.SchemeGroupVersion, storage: storage}, nil
 }
 
+func (c *completedConfig) withAggregator(delegatedAPIServer genericapiserver.DelegationTarget) (*genericapiserver.GenericAPIServer, error) {
+	aggregatorConfig, err := originaggregator.CreateAggregatorConfig(
+		*c.GenericConfig.Config,
+		c.ExtraConfig.KubeInformers,
+		aggregatorapiserver.NewClusterIPServiceResolver(c.ExtraConfig.KubeInformers.Core().V1().Services().Lister()),
+		utilnet.SetTransportDefaults(&http.Transport{}),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	completedConfig := aggregatorConfig.Complete()
+	openApiConfig := completedConfig.GenericConfig.OpenAPIConfig
+	completedConfig.GenericConfig.OpenAPIConfig = nil
+
+	genericServer, err := completedConfig.GenericConfig.New("openapi-aggregator", delegatedAPIServer)
```

Comment:
isn't this still a full server?

Reply:
Per our discussion (and to prevent introducing upstream changes in this PR) I am wrapping the openshift apiserver in a new generic apiserver that has no openapi fields set, in order for the openapi aggregation to work

Reply:
I've discussed with Juan this on Slack a bit more, he'll try to create a simpler approach not setting up full server.